UPGRADE.rst

```diff
@@ -55,50 +55,140 @@ Upgrading to v1.4.0
+
+In the medium term the vector.im and matrix.org identity servers will disable
+sending email and SMS entirely, however in order to ease the transition they
+will retain the capability to send email and SMS for a limited period. Email
+will be disabled on XXX (giving 2 months notice), disabling SMS will follow
+some time after that once SMS sending support lands in Synapse.
+
+Once email and SMS support has been disabled in the vector.im and matrix.org
+identity servers, all Synapse versions that depend on those instances will be
+unable to send email and SMS through them. There are no imminent plans to
+remove email and SMS support from Sydent altogether.
+
+Email
+-----
+
+Following upgrade, to continue using email as a registration method admins can
+either:-
+
+* Configure Synapse to use an alternate email server (details follow).
+* Run or choose an identity server (with email support) and delegate to it.
+
+To configure an SMTP server for Synapse, modify the configuration section
+headed ``email``, and be sure to have at least the ``smtp_host, smtp_port``
+and ``notif_from`` fields filled out.
+
+You may also need to set ``smtp_user``, ``smtp_pass``, and
+``require_transport_security``.
+
+See the `sample configuration file
+<https://github.com/matrix-org/synapse/blob/master/docs/sample_config.yaml>`_
+for more details on these settings.
+
+Some admins will wish to continue using email as a registration method, but
+will not immediately have an appropriate SMTP server to hand.
+
+To this end, we will continue to support email delegation via the vector.im and
+matrix.org identity servers for two months, the cut off date is XXXX after
+which time email delegation will be disabled.
+
+The ``account_threepid_delegates`` dictionary defines whether the homeserver
+should delegate an external server (typically an `identity server
+<https://matrix.org/docs/spec/identity_service/r0.2.1>`_) to handle sending
+password reset or registration messages via email and SMS.
+
+So to delegate email sending set ``account_threepid_delegates.email`` to a base
+URL of an identity server in your homeserver.yaml. Note that
+``account_threepid_delegates.email`` replaces the deprecated
+``email.trust_identity_server_for_password_resets``.
+
+.. code:: yaml
+
+   account_threepid_delegates:
+       email: https://example.com     # Delegate email sending to example.com
 
 If ``email.trust_identity_server_for_password_resets`` is set to ``true``, and
 ``account_threepid_delegates.email`` is not set, then the first entry in
-``trusted_third_party_id_servers`` will be used as the account threepid delegate for email.
-This is to ensure compatibility with existing Synapse installs that set up external server
-handling for these tasks before v1.4.0. If ``email.trust_identity_server_for_password_resets``
-is ``true`` and no trusted identity server domains are configured, Synapse will throw an error.
+``trusted_third_party_id_servers`` will be used as the account threepid
```

Comment:
I think this might be clearer if we replace "account threepid delegate" with ``account_threepid_delegate``